### PR TITLE
Update the wagmi and rainbowkit version

### PR DESCRIPTION
**Pull request by LeifuChen**
## Description
Update both the `wagmi` and `rainbowkit `versions to the latest version.

Note:
1. The `watch-mode` bug in `wagmi` has been fixed in v0.8.5 (ref: https://github.com/wagmi-dev/wagmi/pull/1282). This makes our dapp more smoothly.
2. The old `safe-appd-wagmi` connector is not incompatible with the latest `wagmi` version.
We fixed this by copying the code directly into our code repo. Here is the reference: https://github.com/safe-global/safe-apps-sdk/issues/395
3. #1545 has been fixed after updating the latest rainbowkit version.

## Related issue
#1545 

## Motivation and Context
1. To make the development more productive by increasing the local dev performance
2. To prepare for adding more custom wallets by updating the latest version of rainbowkit.

### CODE REVIEW

File: hooks/useStakingData.ts [hunk 1]
```diff
@@ -212,28 +218,27 @@ const useStakingData = () => {
...
+	const { data: vestingSchedules } = useContractRead({
 		...rewardEscrowContract,
 		functionName: 'getVestingSchedules',
 		args: [walletAddress ?? undefined, 0, 1000],
-		watch: false,
+		scopeKey: 'staking',
+		watch: true,
 		enabled: !!walletAddress,
-		select: (data) => data.filter((d) => d.escrowAmount.gt(0)),
+		select: (data) => (data as VestingEntry[]).filter((d) => d.escrowAmount > 0),
```
Comment (avclarke) [inline hooks/useStakingData.ts:228]: Is this definitely a number? Seems like it would be a BigNumber

File: hooks/useStakingData.ts [hunk 2]
```diff
@@ -50,54 +50,59 @@ type VestingEntry = {
 	entryID: number;
 };
 
+type VestingClaimable = {
+	quantity: number;
```
Comment (avclarke) [inline hooks/useStakingData.ts:54]: Seems like values which should be strings or BigNumbers?

**Review by avclarke — COMMENTED** [review]
Looks good, just a couple of questions on the types.

File: hooks/useStakingData.ts (hunk 1 above)
Comment (LeifuChen) [reply hooks/useStakingData.ts:228]: Yes. This should be `BigNumber`. The change has been made.

File: hooks/useStakingData.ts (hunk 2 above)
Comment (LeifuChen) [reply hooks/useStakingData.ts:54]: Yes. This should be `BigNumber`. The change has been made.

**Review by platschi — APPROVED** [review]

**Pull request merged**